app/uk/gov/hmrc/tai/viewModels/TaxCodeDescriptor.scala

```diff
@@ -29,80 +30,82 @@ trait TaxCodeDescriptor {
 
   case class TaxCodeDescription(taxCode: String, basisOperation: BasisOperation, scottishTaxRateBands: Map[String, BigDecimal])
 
-  def describeTaxCode(taxCode: String, basisOperation: BasisOperation, scottishTaxRateBands: Map[String, BigDecimal])
+  def describeTaxCode(taxCode: String, basisOperation: BasisOperation, scottishTaxRateBands: Map[String, BigDecimal], currentTaxCode: Boolean = true)
               (implicit messages: Messages): ListMap[String, String] = {
 
-    val explanationRules: Seq[TaxCodeDescription => ListMap[String, String]] = Seq(
-      scottishTaxCodeExplanation,
-      untaxedTaxCodeExplanation,
-      fetchTaxCodeExplanation,
-      emergencyTaxCodeExplanation
+    val previousOrCurrent = if (currentTaxCode) "" else ".prev"
```

Comment:
Could this be moved down into the explanation defs, where the rest of the logic for constructing these is? I know it would mean some duplication but possibly more readable?

Reply:
Agreed, it's done now